ramls/permission.json

```diff
@@ -1,58 +1,82 @@
+  "$schema" : "http://json-schema.org/draft-04/schema#",
+  "title" : "Permissions Definition Schema",
+  "type" : "object",
+  "description" : "A Folio permission",
+  "properties" : {
+    "permissionName" : {
+      "description" : "The unique name of the permission",
+      "type" : "string"
+    },
+    "displayName" : {
+      "description" : "The human-readable name of the permission",
+      "type" : "string"
+    },
+    "id" : {
+      "description" : "The globally unique (UUID) identifier for the permission",
+      "type" : "string"
+    },
+    "description" : {
+      "description" : "A description of the permission",
+      "type" : "string"
+    },
+    "tags" : {
+      "description" : "A list of tags for searching",
+      "type" : "array",
+      "items" : {
+        "type" : "string"
       }
     },
-    "subPermissions": {
-      "description": "A list of child permissions granted by this permission",
-      "type": "array"
+    "subPermissions" : {
+      "description" : "A list of child permissions granted by this permission",
+      "type" : "array"
     },
-    "childOf": {
-      "description": "A list of permissions that 'own' this permission",
-      "type": "array"
+    "childOf" : {
+      "description" : "A list of permissions that 'own' this permission",
+      "type" : "array"
     },
-    "grantedTo": {
-      "description": "A list of permission users to whom this permission has been granted",
-      "type": "array"
+    "grantedTo" : {
+      "description" : "A list of permission users to whom this permission has been granted",
+      "type" : "array"
     },
-    "mutable": {
-      "description": "Whether or not this permission changes at runtime",
-      "type": "boolean"
+    "mutable" : {
+      "description" : "Whether or not this permission changes at runtime",
+      "type" : "boolean"
     },
     "visible" : {
-      "description": "Whether or not this permission is visible to users",
+      "description" : "Whether or not this permission is visible to users",
       "type" : "boolean"
     },
     "dummy" : {
-      "description": "Is this a temporary dummy permission",
+      "description" : "Is this a temporary dummy permission",
       "type" : "boolean"
     },
+    "deprecated" : {
+      "description" : "Indicates whether this permissions has been marked for deletion (soft deleted)",
+      "type" : "boolean",
+      "default" : "false"
+    },
+    "defined" : {
+      "description" : "Indicates how this permission was created",
+      "type" : "string",
+      "enum" : [
+        "User",
+        "System"
+      ],
+      "readOnly" : true
+    },
+    "moduleName" : {
+      "description" : "The name of the module (not including version) that defined this permission",
+      "type" : "string",
+      "readOnly" : true
+    },
+    "moduleVersion" : {
+      "description" : "The version of the module that defined this permission",
+      "type" : "string",
+      "readOnly" : true
+    },
```

Comment:
That's not my recollection.  I thought we decided to keep defined(By) and flatten the three props (move them up to Permission level), accepting that there was some overlap between `mutable` and `defined`, but since we couldn't rely on mutable as a reliable indicator that a module was defined by the system (because it wasn't locked down as read-only), this was necessary.

Reply:
defined/definedBy was removed.  Now Permission just has moduleName and moduleVersion.